File: FloodFilling.py
M = 8
N = 8


def Floodfilling(screen, x, y, OldColor, NewColor):
    if (x < 0 or x >= M or y < 0 or
            y >= N or screen[x][y] != OldColor or
            screen[x][y] == NewColor):
        return

    screen[x][y] = NewColor

    Floodfilling(screen, x + 1, y, OldColor, NewColor)
    Floodfilling(screen, x - 1, y, OldColor, NewColor)
    Floodfilling(screen, x, y + 1, OldColor, NewColor)
    Floodfilling(screen, x, y - 1, OldColor, NewColor)

def floodFill(screen, x, y, NewColor):
    OldColor = screen[x][y]
    if (OldColor == NewColor):
        return
    Floodfilling(screen, x, y, OldColor, NewColor)


screen = [[1, 1, 1, 1, 1, 1, 1, 1],
          [1, 1, 1, 1, 1, 1, 0, 0],
          [1, 0, 0, 1, 1, 0, 1, 1],
          [1, 2, 2, 2, 2, 0, 1, 0],
          [1, 1, 1, 2, 2, 0, 1, 0],
          [1, 1, 1, 2, 2, 2, 2, 0],
          [1, 1, 1, 1, 1, 2, 1, 1],
          [1, 1, 1, 1, 1, 2, 2, 1]]

x = 4
y = 4
NewColor = 3
floodFill(screen, x, y, NewColor)

print("After call to floodFill:")
for i in range(M):
    for j in range(N):
        print(screen[i][j], end=' ')
    print()

File: FloodFilling.cpp
#include<iostream.h>
    #include<graphics.h>
    #include<dos.h>
     
    void Floodfilling(int x,int y,int oldcolor,int newcolor){
        if(getpixel(x,y) == oldcolor){
            putpixel(x,y,newcolor);
            Floodfilling(x+1,y,oldcolor,newcolor);
            Floodfilling(x,y+1,oldcolor,newcolor);
            Floodfilling(x-1,y,oldcolor,newcolor);
            Floodfilling(x,y-1,oldcolor,newcolor);
        }
    }
     
    int main(){
        int gm,gd=DETECT,radius;
        int x,y;
        cout<<"Enter x and y positions for circle: "<<endl;
        cin>>x>>y;
        cout<<"Enter radius of circle"<<endl;
        cin>>radius;
        initgraph(&gd,&gm,"c:\\turboc3\\bgi");
        circle(x,y,radius);
        Floodfilling(x,y,0,15);
        delay(3000);
        closegraph();
        return 0;
    }
